lib/src/scheduler.dart

```diff
@@ -0,0 +1,48 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:ffi';
+import 'dart:isolate';
+
+import 'native/realm_core.dart';
+
+final _receivePortFinalizer = Finalizer<RawReceivePort>((p) => p.close());
```

Comment:
If we are running in non-root isolate, and isolate shuts down, then scheduler will be reaped, and port closed by this finalizer.

Reply:
So the `RawReceivePort` finalizer doesn't close the port? Also, what happens in native to any objects that hold a ref to the scheduler?

Reply:
No. That is why `_receivePortFinalizer` is there. And even that will only be called, if the attached object is GC'ed. During process shutdown, dart makes no such promises for non-native finalizers, hence this issue. 

If the scheduler is GC'ed (which can at the earliest happen during isolate shutdown, if at all (as is)), then the `SchedulerHandler` is also garbage, and the native side finalizer for this will run essentially calling `realm_release`.